test/e2e/upgrade/upgrade.go

```diff
@@ -359,10 +359,40 @@ func clusterUpgrade(f *framework.Framework, c configv1client.Interface, dc dynam
 
 	kubeClient := kubernetes.NewForConfigOrDie(config)
 
-	// this is very long.  We should update the clusteroperator junit to give us a duration.
-	maximumDuration := 150 * time.Minute
-	baseDurationToSoftFailure := 75 * time.Minute
-	durationToSoftFailure := baseDurationToSoftFailure
+	const (
+		OVN = "OVNKubernetes"
+		SDN = "OpenShiftSDN"
+	)
+
+	// Maximum duration is our maximum polling interval for upgrades
+	var maximumDuration = 150 * time.Minute
+
+	// defaultUpgradeLimit is the default for platforms not listed below.
+	var defaultUpgradeLimit = 100.0 * time.Minute
+
+	// The durations below were calculated from the P99 gathered from sippy's record of upgrade
+	// durations[1]. We are only considering the tuple of platform and network type, not upgrade
+	// type.  Micro upgrades are volatile because you may only get 1 or 2 components, or you may
+	// get them all. The worst case of a micro upgrade is where all components get updated including
+	// OS, which would be the same as a minor upgrade, so we use the minor threshold for all values.
+	//
+	// [1] https://raw.githubusercontent.com/openshift/sippy/f546164c18db9a4a930cc64f43bcdcd35509e767/scripts/sql/test-duration-stats.sql
+	type limitLocator struct {
+		Platform configv1.PlatformType
+		Network  string
+	}
+	var upgradeDurationLimits = map[limitLocator]float64{
+		{configv1.AWSPlatformType, OVN}:       85,
+		{configv1.AWSPlatformType, SDN}:       130,
```

Comment:
This value worries me, it's thrown off by just a small number of outliers. Here's the by-day stats for aws upgrade-minor, sdn:

```
 timestamp  |   percentile_50   |   percentile_99    |   percentile_100   
------------+-------------------+--------------------+--------------------
 2023-04-03 | 66.33668445480001 |  68.67009867101666 |  68.67009867101666
 2023-04-04 |    61.00122070645 |  65.33670738831667 |  65.33670738831667
 2023-04-05 | 65.66994640008333 |  66.83663089001666 |  66.83663089001666
 2023-04-06 |     71.1699914034 | 130.00100292186667 | 130.00100292186667
 2023-04-07 | 63.50128090998333 |  64.00166007301667 |  64.00166007301667
 2023-04-08 | 64.50178690341667 |  67.83420327996666 |  67.83420327996666
 2023-04-09 | 63.00180272506667 | 63.834119744966664 | 63.834119744966664
 2023-04-10 |    61.83355659865 |  62.50341595853333 |  62.50341595853333
 2023-04-11 | 62.50338116806667 |  63.66693479318334 |  63.66693479318334
 2023-04-12 | 62.83366054938333 |  66.83653658128334 |  66.83653658128334
 2023-04-13 | 65.50332776381667 | 123.83383404793334 | 123.83383404793334
 2023-04-14 | 62.33785857713334 |      63.0011904242 |      63.0011904242
 2023-04-15 | 62.16696633663333 |  65.17051039851667 |  65.17051039851667
 2023-04-16 | 62.50331522281667 |  65.33356643743333 |  65.33356643743333
 2023-04-17 | 61.83411498416667 |  64.16791090768334 |  64.16791090768334
 2023-04-18 | 60.66809686411667 |  63.83653952776667 |  63.83653952776667
 2023-04-19 | 63.50436424363333 |  67.67001490021667 |  67.67001490021667
 2023-04-20 | 67.83669018163333 |  70.16891166096666 |  70.16891166096666
 2023-04-21 | 68.83359635573333 |  71.33662739818332 |  71.33662739818332
 2023-04-22 |    68.00084324615 |  72.83777085256666 |  72.83777085256666
 2023-04-23 |     66.8351065497 |  71.33723795998334 |  71.33723795998334
 2023-04-24 |     68.8345876203 |  70.00339408296666 |  70.00339408296666
 2023-04-25 |    68.67053672035 |     71.67106336005 |     71.67106336005
 2023-04-26 | 66.66754525656667 |  72.83672768596666 |  72.83672768596666
 2023-04-27 | 67.16839730008333 | 133.50677713456668 | 133.50677713456668
 2023-04-28 | 69.66994769221667 |  71.33660977888333 |  71.33660977888333
 2023-04-29 | 68.83508444216666 |  70.00376385868334 |  70.00376385868334
 2023-04-30 | 67.50338064378333 |  71.33662319876666 |  71.33662319876666
 2023-05-01 | 69.67003445261666 |  74.16994683001667 |  74.16994683001667
 2023-05-02 | 69.16996374988334 |  70.00433161243333 |  70.00433161243333
 2023-05-03 | 65.00124213528333 |  67.66696476931666 |  67.66696476931666
```

There's a handful of days (not clustered together) where the p99 is >120.  If we throw them away, we get a value around 75 which seems to make more sense to me.